parsing/parsetree.mli

```diff
@@ -337,8 +337,8 @@ and expression_desc =
       [C] represents a type constraint or coercion placed immediately before the
       arrow, e.g. [fun P1 ... Pn : ty -> ...] when [C = Some (Pconstraint ty)].
 
-      A function must have parameters. [Pexp_function (params, _, body)] must
-      have non-empty [params] or a [Pfunction_cases _] body.
+      A function must have parameters: in [Pexp_function (params, _, body)],
+      [params] contains a [Pparam_val _] or body is [Pfunction_cases _].
```

Comment:
If the intended fix is that `params` must contain a `Pparam_val _`, maybe we should als avoid the natural language ambiguity between `or` and `xor`:

> if [params] does not contain a [Pparam_val _], body must be [Pfunction_cases _]

?

Reply:
Sure.